app/forms/openid_connect_authorize_form.rb

```diff
@@ -62,7 +62,7 @@ def loa3_requested?
 
   def allowed_form_action
     return unless sp_redirect_uri =~ %r{https?://}
-    SecureHeadersWhitelister.extract_domain(sp_redirect_uri)
+    SecureHeadersWhitelister.extract_scheme_and_domain(sp_redirect_uri)
```

Comment:
If I understand that correctly, there's no harm in us putting in the full `redirect_uri` URL in `form-action`, which saves us the trouble of extracting the protocol + domain.